lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -379,39 +385,67 @@ class PropagateAtenSliceTensorPattern
     if (!matchPattern(op.getStep(), m_TorchConstantInt(&step)))
       return rewriter.notifyMatchFailure(op, "requires a constant step");
 
-    if (step < 0)
-      return rewriter.notifyMatchFailure(op, "requires a positive step value");
-
-    auto selfTy = cast<BaseTensorType>(op.getSelf().getType());
     auto selfShape = selfTy.getSizes();
+    auto resultShape = resultTy.getSizes();
     int64_t selfRank = selfShape.size();
 
     // Correct for negative indexing:
     dim = dim < 0 ? dim + selfRank : dim;
 
-    int64_t dimLength = elements.size();
+    int64_t dimLength = selfShape[dim];
```

Comment:
Could you use `toPositiveDim` + `isValidDim` here as well (also similar problem with PropagateIndexSelectPattern)? 

Since this pass is matching against constants not known during verification, a lot of UB in the input MLIR will go undetected and could cause compiler crashes. So, it probably makes sense to have strict checks whenever there is a `matchPattern`